agent/cache-types/catalog_list_services.go

```diff
@@ -38,15 +38,23 @@ func (c *CatalogListServices) Fetch(opts cache.FetchOptions, req cache.Request)
 	// going to be served from cache and end up arbitrarily stale anyway. This
 	// allows cached service-discover to automatically read scale across all
 	// servers too.
-	reqReal.AllowStale = true
```

Comment:
I don't get this request.AllowStale is not used anymore?

Reply:
Agreed, all cache-type RPCs should be sent with AllowStale set to true.

Reply:
I don't understand the question. I did not remove `AllowStale`. I change it to use the full path to the field, instead of the embedded path.

```go
reqReal.QueryOptions.AllowStale = true
```

Reply:
yup sorry,  my bad